lib/ddtrace/transport/http/adapters/ethon.rb

```diff
@@ -0,0 +1,98 @@
+require 'ddtrace/transport/response'
+require 'ethon'
+
+module Datadog
+  module Transport
+    module HTTP
+      module Adapters
+        # Adapter for Ethon >= 0.8.0.
+        # More memory efficient than Net::HTTP.
+        class Ethon
+          attr_reader \
+            :hostname,
+            :port,
+            :timeout
+
+          DEFAULT_TIMEOUT = 10
+
+          def initialize(
+            hostname = Datadog::Transport::HTTP.default_hostname,
+            port = Datadog::Transport::HTTP.default_port,
+            options = {}
+          )
```

Comment:
If I'm reading this correctly, by having the hostname, port and options here, we expose them to being called using `t.adapter` in the configuration proc.

I'm somewhat confused by the advantage of doing this -- why would a customer want to use a custom hostname/port at the transport level, rather than just configuring it using the usual `Datadog.configure { |c| c.tracer.hostname = ... }`?

I understand why we need it internally for testing and for separation, but have you considered not exposing it externally? It would be cool if we could mostly centralize these settings so that there would be one way (two, if we count environment variables) of configuring these settings :)

Reply:
I wondered about this question while looking at the `GettingStarted.md`:

> #### Using the Ethon adapter
> [...]
> ```ruby
> Datadog.configure do |c|
>   c.tracer.transport_options = proc { |t|
>     t.adapter :ethon
>  }
> end
> ```
>
> #### Using the Net::HTTP adapter
> [...]
> ```ruby
> Datadog.configure do |c|
>   c.tracer.transport_options = proc { |t|
>     # Hostname, port, and additional options. :timeout is in seconds.
>     t.adapter :net_http, '127.0.0.1', 8126, { timeout: 1 }
>   }
> end
> ```

I realized that these arguments were actually being exposed to customers (even if Marco was not documenting them for ethon, like we have for http), and that got me wondering what the value of a transport-specific override to the regular ddtrace gem was.